nativelib/src/main/resources/scala-native/gc/shared/ScalaNativeGC.h

```diff
@@ -12,12 +12,13 @@
 #pragma comment(lib, "kernel32.lib")
 #include <windows.h>
 typedef DWORD ThreadRoutineReturnType;
+typedef ThreadRoutineReturnType(WINAPI *ThreadStartRoutine)(void *);
 #else
 #include <pthread.h>
 typedef void *ThreadRoutineReturnType;
+typedef ThreadRoutineReturnType (*ThreadStartRoutine)(void *);
```

Comment:
I think we can define a no-op `WINAPI` when not using Windows and then use a single definition for easier maintenance in the future. Seems like we might need more of them in the glue layer anyway.

Reply:
Let me check because I think this is only needed for Windows 32-bit which is the default when using the MSVC toolchain.

Reply:
It seems that the Visual Studio 2022 compiler defaults to 32 bit which requires this extra `WINAPI` thing or it won't compile. 64 bit ignores the `WINAPI`. I still haven't managed to get the native toolchain to link but I don't think this is a terrible change as at least the code compiles. There is just 2 branches, one for Windows and one for the rest so I am not sure about what you are asking above.

I really think we should be able to compile via the Windows tools but I haven't been able to decipher the linking issue yet - lack of time and knowledge at this point.

Reply:
@WojciechMazur I made an attempt to fix this like you suggested but I am not sure what type to make the typedef or perhaps `#define` might work. I will experiment on macOS to see if I can figure it out there (works on WIndows). Any suggestions appreciated.

Reply:
Ok, the latest may not be 100% bulletproof but is at least workable.

Reply:
Sorry for no reply previously. I was referring to `WINAPI` defined using `#define` which on non-Windows machines would be simply `#define WINAPI`. 
I believe any solution should be fine. The question is can we test it somehow? On the current GitHub Actions setup there was no issue with that so far, but I guess these error exists in local environments

Reply:
Sure I can try that - originally I did a typedef which wouldn't work. This is just a compile error - maybe we can set up to test Windows using built in clang. Let me try some things. The main problem is the second checkbox above - the linking issues.